src/commands/luckymon.rs

```diff
@@ -37,6 +37,72 @@ async fn read_daily_entry(ctx: &Context, user_id: UserId) -> Option<(i64, bool)>
     entry
 }
 
+fn capitalize(name: &str) -> String {
+    let (n1, n2) = name.split_at(1);
+    let n_upper = n1.to_uppercase();
+    let new_name = n_upper + n2;
+    return new_name.to_string();
+}
+
+fn capitalize_hyphenated(name: &str, keep_hyphen: bool) -> String {
+    let hyphen_index = name.chars().position(|c| c == '-').unwrap();
+    let (n1, eh) = name.split_at(hyphen_index);
+    let (_uh, n2) = eh.split_at(1);
+    let middle_char = if keep_hyphen { "-" } else { " " };
+    let new_name = capitalize(n1) + middle_char + &capitalize(n2);
+    return new_name.to_string();
+}
+
+
+fn has_hyphen(name: &str) -> bool {
+    return name.contains("-");
+}
+
+fn is_nidoran(name: &str) -> bool {
```

Comment:
I was kind of torn on this.  Technically we could pass the actual pokemon in and check the numbers (female and male being 29 and 32, respectively, and we could check the same for porygon z and ho-oh), but I had already started passing just the species name around everywhere.  Happy to refactor to something like that though!